src/main/java/org/generationcp/middleware/api/breedingmethod/BreedingMethodServiceImpl.java

```diff
@@ -44,4 +46,27 @@ public BreedingMethodDTO getBreedingMethod(final Integer breedingMethodDbId) {
 		final Method methodEntity = this.daoFactory.getMethodDAO().getById(breedingMethodDbId);
 		return new BreedingMethodDTO(methodEntity);
 	}
+
+	@Override
+	public List<BreedingMethodDTO> getBreedingMethods(final String programUUID, final boolean favorites) {
+		final List<Method> breedingMethods;
+		if (!StringUtils.isEmpty(programUUID) && favorites) {
+			final List<Integer> breedingMethodIds = this.getFavoriteProjectMethodsIds(programUUID);
+			breedingMethods = this.daoFactory.getMethodDAO().getMethodsByIds(breedingMethodIds);
+		} else {
+			breedingMethods = this.daoFactory.getMethodDAO().getAllMethod();
```

Comment:
Almost sure that this will not cover the case where an user created a method for a program and then he wants to know all the breeding methods, I think this query will exclude the ones that were created for an specific program